### ISSUE TITLE
[BUG] `ForecastingHorizon.to_relative` wrongly converts `pd.index` with frequencies that are base frequency multiples to `int`s

### ISSUE THREAD

**Issue by tomshlomo** [alan-turing-institute/sktime#1935]
**Describe the bug**

`ForecastHorizon.to_relative` is wrong when using pandas date ranges with frequencies that are multiples of base frequencies (e.g "2H", "30T", "10D").
I believe the cause is the function `_coerce_duration_to_int`.

Seems to be related to #1435 and #1876 (sorry if this is a duplicate)

**To Reproduce**

```python
import pandas as pd
from sktime.forecasting.base import ForecastingHorizon

freq = '2H'
t = pd.date_range(start='2021-01-01', freq=freq, periods=5)
fh_abs = ForecastingHorizon(t, is_relative=False)
fh_rel = fh_abs.to_relative(cutoff=t.min())
print(fh_rel)
```
Output:
```
ForecastingHorizon([0, 4, 8, 12, 16], dtype='int64', is_relative=True)
```

**Expected behavior**

Output should be:
```
ForecastingHorizon([0, 1, 2, 3, 4], dtype='int64', is_relative=True)
```
no matter what `freq` is.

To see this bug in action, consider the following example of using a NaiveForecaster to predict a perfectly periodic time series:
```python
import pandas as pd
import numpy as np
from sktime.forecasting.base import ForecastingHorizon
from sktime.forecasting.model_selection import temporal_train_test_split
from sktime.forecasting.naive import NaiveForecaster

sp = 5  # period
n = sp * 2  # number of samples

y = pd.Series(
    index=pd.date_range(start='2021-01-01', freq='2H', periods=n),
    data=np.mod(np.arange(n), sp),
)

y_train, y_test = temporal_train_test_split(y, test_size=sp)
fh = ForecastingHorizon(y_test.index, is_relative=False)
forecaster = NaiveForecaster(strategy="last", sp=sp)
forecaster.fit(y_train)
y_pred = forecaster.predict(fh)
print(y_test, y_pred, np.abs(y_pred - y_test).max(), sep='\n\n')
```

Output:
```
2021-01-01 10:00:00    0
2021-01-01 12:00:00    1
2021-01-01 14:00:00    2
2021-01-01 16:00:00    3
2021-01-01 18:00:00    4
Freq: 2H, dtype: int64

2021-01-01 10:00:00    3.0
2021-01-01 12:00:00    2.0
2021-01-01 14:00:00    1.0
2021-01-01 16:00:00    0.0
2021-01-01 18:00:00    4.0
dtype: float64

3.0
```
**Versions**
<details>

Python 3.9.5 (default, May 18 2021, 12:31:01) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.0.1 -- An enhanced Interactive Python. Type '?' for help.
PyDev console: using IPython 8.0.1
System:
    python: 3.9.5 (default, May 18 2021, 12:31:01)  [Clang 10.0.0 ]
executable: /Users/tomshlomo/miniconda3/envs/playground-395/bin/python
   machine: macOS-10.15.7-x86_64-i386-64bit
Python dependencies:
          pip: 21.1.1
   setuptools: 52.0.0.post20210125
      sklearn: 1.0.2
       sktime: 0.9.0
  statsmodels: 0.13.1
        numpy: 1.20.2
        scipy: 1.6.2
       Cython: 0.29.26
       pandas: 1.2.4
   matplotlib: 3.3.4
       joblib: 1.0.1
        numba: 0.54.1
     pmdarima: None
      tsfresh: 0.17.0
</details>

**Comment by fkiraly** [comment]
> Seems to be related to #1435 and #1876 (sorry if this is a duplicate)

Not sure if it's a duplicate, but definitely seems related and may have the same root cause.
FYI @eenticott-shell who was looking at the other two.

**Comment by eenticott-shell** [comment]
I will investigate this as well, they do all seem to be related.

**Comment by fkiraly** [comment]
> I will investigate this as well, they do all seem to be related.

By now, I assume your mindmap on this must look like that stereotypical crazy conspiracy theorist pinboard with newspaper clippings and red string? :-)

**Comment by eenticott-shell** [comment]
There seems to be two parts to this bug that I have identified. The first seems to be a problem in pandas. To find the relative horizon the cutoff is taken away from the absolute horizon. To achieve this they are first converted through `to_period()`. The type of output from `to_period` is a `Period` for the cutoff timestamp and a `PeriodIndex` for the `DateTimeIndex`. I suspect the differing types cause an issue in pandas. When one of these is taken from the other, the interval is multiplied by the frequency, hence with 2 hours you get [0, 4, 8, ...] etc. With a frequency of 3H this becomes [0, 9, 18, ...]. However, if I manually create both the cutoff and absolute horizon as `PeriodIndex` then the result is [0, 2, 4, ...] which is the expected result for this part. 

The second problem is in the `_coerce_duration_to_int` where it takes the output of the above and just extracts the integers seemingly ignoring the frequency. I haven't investigated this too thoroughly yet. I will work on fixing this.

**Comment by fkiraly** [comment]
@eenticott-shell, could you kindly post minimally reproducing code for the issues? The code I've seen for no.1 narrows it down to `pandas`.

**Comment by eenticott-shell** [comment]
@fkiraly This shows the difference in output between the two types. As far as I am aware these should both produce the same result:

```{python}
import pandas as pd
absolute = pd.PeriodIndex(['2021-01-01 00:00', '2021-01-01 03:00', '2021-01-01 06:00',
             '2021-01-01 09:00', '2021-01-01 12:00'], freq = '3H')

relativec_as_PI = pd.PeriodIndex(['2021-01-01 00:00'], freq = '3H')

print(absolute - relativec_as_PI)

relative_as_P = pd.Period('2021-01-01 00:00', freq = '3H')

print(absolute - relative_as_P)
```

**Comment by khrapovs** [comment]
Same creepiness is happening in https://github.com/alan-turing-institute/sktime/issues/805

**Comment by fkiraly** [comment]
Yes, this looks like a `pandas` bug. We should report it (or check whether it is already known) and link it here.
@eenticott-shell, have you done that, or would you prefer me to do it?

**Comment by eenticott-shell** [comment]
I have not yet reported it to `pandas` but I am happy to do this. I do believe we will be able to sidestep the bug without having to wait for `pandas` to fix it though.

**Comment by fkiraly** [comment]
This should be fixed by #1981.

**Comment by tomshlomo** [comment]
Amazing, thank you very much!
This fix is didn't get in 0.10 right?

**Comment by fkiraly** [comment]
no, sorry.
Will be in the next release though - and you can test it on the dev version (that would be appreciated in case we overlooked somthing)
